src/formatDate.js

```diff
@@ -50,7 +50,54 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  function changeYearFormat(year, requiredLength) {
+    if (year.length === requiredLength) {
+      return year;
+    }
+
+    if (requiredLength === 2) {
+      return year.slice(2);
+    }
+
+    return year < 30 ? '20' + year : '19' + year;
+  }
+
+  const dateSplited = date.split(fromFormat[3]);
+  const newDate = {};
+
+  for (let i = 0; i < dateSplited.length; i++) {
+    switch (fromFormat[i]) {
+      case 'YY':
+      case 'YYYY':
+        newDate.year = dateSplited[i];
```

Comment:
As a variant, you may have not created fields 'year', 'month', and 'day' but just use 'DD', 'YY' or 'MM' as keys in your `newDate` object:
```
  for (let i = 0; i < fromFormat.length - 1; i++) {
    newDate[fromFormat[i]] = dateSplited[i];
  }
```
But your variant is ok too 👍